stabilizer/src/main/java/com/hazelcast/stabilizer/provisioner/AwsProvisioner.java

```diff
@@ -0,0 +1,333 @@
+        }
+        return foundInstances;
+    }
+
+    public boolean waiteForInstanceStatusRunning(Instance i){
+
+        DescribeInstancesRequest describeInstancesRequest = new DescribeInstancesRequest().withInstanceIds(i.getInstanceId());
+        int counter=0;
+        while ( counter++ < MAX_SLEEPING_ITTERATIONS ) {
+            sleepMillis(SLEEPING_MS);
+
+            DescribeInstancesResult result = ec2.describeInstances(describeInstancesRequest);
+
+            for ( Reservation r : result.getReservations() ) {
+                for ( Instance j : r.getInstances() ) {
+                    if ( j.getPublicIpAddress() != null  && j.getState().getName().equals(AWS_RUNNING_STATE) ) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    public void addInstanceToAgentsFile(Instance i){
+        DescribeInstancesRequest describeInstancesRequest = new DescribeInstancesRequest().withInstanceIds(i.getInstanceId());
+        DescribeInstancesResult result = ec2.describeInstances(describeInstancesRequest);
+
+        for(Reservation r : result.getReservations() ){
+            for(Instance j : r.getInstances() ){
+                appendText(j.getPublicIpAddress() + "," + j.getPrivateIpAddress() + "\n", agentsFile);
+
+            }
+        }
+    }
+
+    public String createLoadBalancer(String name){
+
+        CreateLoadBalancerRequest lbRequest = new CreateLoadBalancerRequest();
+        lbRequest.setLoadBalancerName(name);
+        lbRequest.withAvailabilityZones(elbAzs.split(","));
+
+        List<Listener> listeners = new ArrayList();
+        listeners.add(new Listener(elbProtocol, elbPortIn, elbPortOut));
+        lbRequest.setListeners(listeners);
+
+        CreateLoadBalancerResult lbResult=elb.createLoadBalancer(lbRequest);
+        lbResult.getDNSName();
+        appendText(lbResult.getDNSName() + "\n", elbFile);
+
+        return lbRequest.getLoadBalancerName();
+    }
+
+    public boolean isBalancerAlive(String name){
+
+        Collection<String> names = new HashSet();
+        names.add(name);
+
+        DescribeLoadBalancersRequest describe = new DescribeLoadBalancersRequest();
+        describe.setLoadBalancerNames(names);
+
+        try{
+            DescribeLoadBalancersResult res=  elb.describeLoadBalancers(describe);
+            List<LoadBalancerDescription> description = res.getLoadBalancerDescriptions();
+
+            if(description.isEmpty()){
+                return false;
+            }
+
+            return true;
+
+        }catch (AmazonServiceException e) {
+            log.fatal(e);
+        }
+        return false;
+    }
+
+    public void addInstancesToElb(String name, List<Instance> instances){
+        if(instances.isEmpty()){
+            log.info("No instances to add to load balance "+name);
+            return;
+        }
+
+        //get instance id's
+        Collection ids = new ArrayList();
```

Comment:
Can you add a type parameter of the type of object you are putting in that ids list.